codecs/avif/enc/avif_enc.cpp

```diff
@@ -49,17 +49,7 @@ val encode(std::string buffer, int width, int height, AvifOptions options) {
   avifRGBImage srcRGB;
   avifRGBImageSetDefaults(&srcRGB, image);
   avifRGBImageAllocatePixels(&srcRGB);
-
-  for (int y = 0; y < height; y++) {
-    for (int x = 0; x < width; x++) {
-      int pixelOffset = y * width + x;
-      uint8_t* pixel = &srcRGB.pixels[(4 * x) + (srcRGB.rowBytes * y)];
-      pixel[0] = rgba[pixelOffset * 4 + 0];
-      pixel[1] = rgba[pixelOffset * 4 + 1];
-      pixel[2] = rgba[pixelOffset * 4 + 2];
-      pixel[3] = rgba[pixelOffset * 4 + 3];
-    }
-  }
+  memcpy(srcRGB.pixels, rgba, width * height * 4);
```

Comment:
This looks better. What I've meant by my last suggestion is that, as the next step after `memcpy`, you can just set `pixels` field to the original pointer - then you don't need to allocate and copy any data at all, and instead AVIF will perform conversion directly from provided source into the YUV format.

One more thing I've noticed is that you call `avifRGBImageAllocatePixels` but not `avifRGBImageFreePixels`. As far as I can tell from a quick glance, this will currently cause a memory leak for the allocated pixels.

It can be fixed by calling the `*Free*` counterpart, but, better yet, if you set `pixels` field as described above, then you can remove `avifRGBImageAllocatePixels` line and this issue will go away too, making code both faster and safer :)

I'd probably recommend trying this out in Squoosh with sanitizer enabled once we are done, but this can be left as part of the original PR.

Reply:
I suspect you'll need to free `output` too like official example does using `avifRWDataFree`.

Reply:
You'll probably also want to perform free/destroy above the `encodeResult` condition, otherwise those structures will get leaked on encoding errors.

Reply:
> What I've meant by my last suggestion is that, as the next step after memcpy, you can just set pixels field to the original pointer

I tried using `rgba` directly, but it made the codec crash. My hunch is that `avifRGBImageAllocatePixels` does more under the hood. I’ll add the `*Free*` counterpart.

Reply:
> My hunch is that `avifRGBImageAllocatePixels` does more under the hood.

Looking at the source, the only thing it does is setting `rowBytes` (which you can set manually).

And option of supplying your own `pixels` is documented as allowed in README, too: https://github.com/AOMediaCodec/libavif/blob/f41bcea5c6ec1a8b1f28833e878ced4468144a79/README.md#L158

Reply:
However, if you want to, you can leave this for later, I'm happy to play with it myself if it's not something of interest to you :)

Reply:
OOoh, I never set `rowBytes`. I’ll try doing that.

Reply:
Yeah that was it 🤦

Reply:
I like that GitHub never bothered to show me your comments until I reloaded.